src/utils/conversion.js

```diff
@@ -0,0 +1,99 @@
+    field: action.selectedField,
+    // Need to convert expression for variable
+    variable: action.variable,
+    value: action.value,
+    softLock: action.softLock,
+    cId: action.cId,
+  };
+  switch (action.actionType) {
+    case 'clearOther':
+      newAction.actionType = 'clearAllButThis';
+      break;
+    case 'unlockAllAndClearAll':
+      newAction.actionType = 'clearAll';
+      newAction.softLock = true;
+      break;
+    case 'selectField':
+      newAction.actionType = 'selectValues';
+      break;
+    case 'selectAndLockField':
+      newProperties.actions.push({ ...newAction, cId: null, actionType: 'selectValues' });
+      newAction.actionType = 'lockField';
+      break;
+    default:
+      break;
+  }
+  return newProperties.actions.push(newAction);
+}
+
+export function convertNavigation(oldType) {
+  switch (oldType) {
+    case 'gotoSheet':
+      return 'goToSheet';
+    case 'gotoSheetById':
+      return 'goToSheetById';
+    case 'gotoStory':
+      return 'goToStory';
+    default:
+      return oldType;
+  }
+}
+
+const importProperties = (exportedFmt, initialProperties) => {
+  const newPropertyTree = { qChildren: [] };
+  const newProperties = {
+    actions: [],
+    // Adding props to avoid errors from old navigation button. Converting to any other chart will still give these errors
+    props: { useEnabledCondition: null, fullWidth: 'auto' },
+    qLayoutExclude: { disabled: {} },
+    ...initialProperties,
+  };
+
+  if (exportedFmt && exportedFmt.properties.visualization === 'qlik-button-for-navigation') {
+    Object.keys(exportedFmt.properties).forEach(key => {
+      let props;
+      switch (key) {
+        case 'qLayoutExclude':
+          break;
+        case 'props':
+          props = exportedFmt.properties[key];
+          newProperties.style = {
+            label: props.buttonLabel,
+            showIcon: props.buttonShowIcon,
+            icon: props.buttonIconLui,
+            textAlign: props.buttonTextAlign,
+          };
+          newProperties.useEnabledCondition = props.useEnabledCondition;
+          newProperties.enabledCondition = props.enabledCondition;
+          props.actionItems &&
+            props.actionItems.forEach(actionItem => {
+              convertAction(actionItem, newProperties);
+            });
+          newProperties.navigation = {
+            action: convertNavigation(props.navigationAction),
+            // Need to convert sheet from expression
+            sheet: props.navigationAction === 'gotoSheetById' ? props.sheetId : props.selectedSheet,
+            story: props.selectedStory,
+            websiteUrl: props.websiteUrl,
+            sameWindow: props.sameWindow,
+          };
+          break;
+        case 'qStateName':
+          newProperties[key] = exportedFmt.properties[key];
+          break;
+        default:
+          newProperties.qLayoutExclude.disabled[key] = exportedFmt.properties[key];
```

Comment:
Stuff that is not used in the new button?

Reply:
Yes, I'm not really sure if this is needed and what it is used for. But the conversions in sense seem to do the same. I think it had something to do with reverting back, but I'm not really sure

Reply:
I'd say ask someone in V&I, or @Caele for that matter

Reply:
yea, it is for when converting back, but in this case you cannot do that so do not transfer over things that are not used.

Reply:
I'll remove it then